src/test/java/org/junitpioneer/jupiter/cartesian/CartesianTestExtensionTests.java

```diff
@@ -353,6 +360,38 @@ void testClassWithConstructor() {
 
 		}
 
+		@Nested
+		@DisplayName("use custom arguments provider")
+		class CartesianProductCustomArgumentsProviderTests {
+
+			@Test
+			@DisplayName("when configured on parameters")
+			void usesCustomCartesianArgumentsProviderOnParameters() {
+				ExecutionResults results = PioneerTestKit
+						.executeTestMethodWithParameterTypes(CustomCartesianArgumentsProviderTestCases.class,
+							"twoCustomCartesianArgumentProviders", String.class, int.class);
+
+				assertThat(results).hasNumberOfDynamicallyRegisteredTests(6).hasNumberOfSucceededTests(6);
+				assertThat(results)
+						.hasNumberOfReportEntries(6)
+						.withValues("first(1)", "first(2)", "second(1)", "second(2)", "third(1)", "third(2)");
+			}
+
+			@Test
+			@DisplayName("when configured on test")
+			void usesCustomArgumentsProviderOnTest() {
+				ExecutionResults results = PioneerTestKit
+						.executeTestMethodWithParameterTypes(CustomCartesianArgumentsProviderTestCases.class,
+							"customArgumentProvider", String.class);
+
+				assertThat(results).hasNumberOfDynamicallyRegisteredTests(6).hasNumberOfSucceededTests(6);
+				assertThat(results)
+						.hasNumberOfReportEntries(6)
+						.withValues("first(1)", "first(2)", "second(1)", "second(2)", "third(1)", "third(2)");
+			}
```

Comment:
According to https://github.com/junit-pioneer/junit-pioneer/issues/520#issuecomment-917619959 this is not yet supported. Which means that we could remove it. I've added this only as a showcase